x/metadata/keeper/query_server.go

```diff
@@ -649,6 +650,38 @@ func (k Keeper) ScopeSpecification(c context.Context, req *types.ScopeSpecificat
 		retval.ScopeSpecification = types.WrapScopeSpecNotFound(specAddr)
 	}
 
+	if found && req.IncludeContractSpecs {
+		err := k.IterateContractSpecs(ctx, func(cs types.ContractSpecification) (stop bool) {
```

Comment:
This is way too much work. Just loop over `spec.ContractSpecIds` and get each one specifically: `k.GetContractSpecification(ctx, contractSpectID)`.

Reply:
I avoided `k.GetContractSpecification(ctx, contractSpectID)` because it uses `MustUnmarshal`, which panics. It's an edge case but I wanted to avoid a panic on a standard query.

Reply:
You can't iterate over all contract specs every time you need a couple. The query just won't work in the long run.

For example on either mainnet or testnet, if you try to get a count of contract specs using `provenanced q metadata all contractspecs --limit 1 --count-total`, the first attempt will time out because there's just too many too loop through all at once. A second attempt can succeed because it's loaded the first chunk of them into a cache. But it might even take three or more. Try getting a count of scopes; I can't remember how many times I had to try before it returned something for me.

We have all the information we need, there's no need to look at everything to find it.

Reply:
changed to use `k.GetContractSpecification` for better performance.